libs/sdk-core/src/persist/transactions.rs

```diff
@@ -283,54 +246,78 @@ impl SqliteStorage {
             )?
             .map(|i| i.unwrap())
             .collect();
-
         Ok(vec)
     }
 
+    pub fn select_payments_query(
+        &self,
+        where_clause: &str,
+        offset: u32,
+        limit: u32,
+    ) -> PersistResult<String> {
+        let swap_fields = self.select_swap_fields("swaps_");
+        let swap_query = self.select_swap_query("true", "swaps_");
+        let rev_swap_fields = self.select_reverse_swap_fields("revswaps_");
+        let rev_swap_query = self.select_reverse_swap_query("true", "revswaps_");
+        let query = format!(
+            "
+          SELECT 
+           p.id,
+           p.payment_type,
+           p.payment_time,
+           p.amount_msat,
+           p.fee_msat,
+           p.status,
+           p.description,
+           p.details,
+           e.lnurl_success_action,
+           e.lnurl_metadata,
+           e.ln_address,
+           e.lnurl_withdraw_endpoint,
+           e.attempted_amount_msat,
+           e.attempted_error,
+           o.payer_amount_msat,
+           o.open_channel_bolt11,
+           m.metadata,
+           e.lnurl_pay_domain,
+           {swap_fields},
+           {rev_swap_fields}
+          FROM payments p
+          LEFT JOIN sync.payments_external_info e
+          ON
+           p.id = e.payment_id
+          LEFT JOIN sync.payments_metadata m
+          ON
+            p.id = m.payment_id
+          LEFT JOIN sync.open_channel_payment_info o
+           ON
+            p.id = o.payment_hash
+          LEFT JOIN ({swap_query}) as swaps
+            ON
+              json_extract(p.details, '$.payment_hash') = hex(swaps_payment_hash) COLLATE NOCASE
```

Comment:
Is it possible to `p.id` here instead?
```suggestion
              p.id = hex(swaps_payment_hash) COLLATE NOCASE
```

Reply:
Yes,  indeed better. I fixed that.